src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -62,6 +59,26 @@ public Artefact createPublication(Artefact artefact, String payload) {
         return artefactRepository.save(artefact);
     }
 
+    public Artefact createPublication(Artefact artefact, MultipartFile file) {
+        checkExisting(artefact);
+
+        String blobUrl = azureBlobService.uploadFlatFile(
+            artefact.getSourceArtefactId(),
+            artefact.getProvenance(),
+            file
+        );
+        artefact.setPayload(blobUrl);
+
+        return artefactRepository.save(artefact);
+    }
+
+    private void checkExisting(Artefact artefact) {
```

Comment:
Could we update the method name on this one to highlight it updates the artefact ID, just so it's clear that the method has a side effect of updating the artefact

Reply:
👍🏼